src/pieces/assets/assets_api.py

```diff
@@ -0,0 +1,235 @@
+from pieces.settings import Settings
+from pieces.gui import show_error
+from typing import Dict,Optional
+import json
+import queue
+
+from pieces_os_client.api.assets_api import AssetsApi
+from pieces_os_client.api.asset_api import AssetApi
+from pieces_os_client.api.format_api import FormatApi
+
+from pieces_os_client.models.seed import Seed
+from pieces_os_client.models.seeded_asset import SeededAsset
+from pieces_os_client.models.seeded_format import SeededFormat
+from pieces_os_client.models.seeded_fragment import SeededFragment
+from pieces_os_client.models.transferable_string import TransferableString
+from pieces_os_client.models.classification_generic_enum import ClassificationGenericEnum
+from pieces_os_client.models.asset_reclassification import AssetReclassification
+from pieces_os_client.models.asset import Asset
+from pieces_os_client.models.format import Format
+from pieces_os_client.models.streamed_identifiers import StreamedIdentifiers
+
+
+class AssetsCommandsApi:
+	assets_snapshot: Dict[str, Optional[Asset]] = {} # should be filled in the run in loop
+	asset_queue = queue.Queue() # Queue for asset_ids to be processed
+	block = True # to wait for the queue to recevive the first asset id
+	asset_set = set()  # Set for asset_ids in the queue
+
+	@staticmethod
+	def create_new_asset(raw_string, metadata=None):
+		assets_api = AssetsApi(Settings.api_client)
+
+		# Construct a Seed
+		seed = Seed(
+			asset=SeededAsset(
+				application=Settings.application,
+				format=SeededFormat(
+					fragment=SeededFragment(
+						string=TransferableString(raw=raw_string)
+					)
+				),
+				metadata=metadata  # This should be constructed as per the SDK's definition
+			),
+			type="SEEDED_ASSET"
+		)
+		
+		# Creating the new asset using the assets API
+		created_asset = assets_api.assets_create_new_asset(transferables=False, seed=seed)
+		return created_asset
+	
+	@classmethod
+	def get_assets_snapshot(cls):
+		if cls.assets_snapshot:
+			return cls.assets_snapshot
+	
+
+		assets_api = AssetsApi(Settings.api_client)
+		# Call the API to get assets identifiers
+		api_response = assets_api.assets_identifiers_snapshot()
+
+		# Extract the 'id' values from each item in the 'iterable' list
+		cls.assets_snapshot = {item.id:None for item in api_response.iterable}
+
+		return cls.assets_snapshot
+		
+	@classmethod
+	def update_asset_snapshot(cls,asset_id):
+		asset_api = AssetApi(Settings.api_client)
+		asset = asset_api.asset_snapshot(asset_id)
+		cls.assets_snapshot[asset_id] = asset # Cache the asset
+		return asset
+
+	@classmethod
+	def worker(cls):
+		try:
+			while True:
+				asset_id = cls.asset_queue.get(block=cls.block,timeout=5)
+				cls.asset_set.remove(asset_id)  # Remove asset_id from the set
+				cls.update_asset_snapshot(asset_id)
+				cls.asset_queue.task_done()
+		except queue.Empty: # queue is empty and the block is false
+			if cls.block:
```

Comment:
what is this cls.block property? want to ensure we don't run into an infinite loop here.

Reply:
It will be set to False when all the assets in the receivered stream identifiers are added to the queue

Reply:
Check line 102